aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -195,7 +195,29 @@ bool is_nonzero(const Tensor& self) {
 
 namespace {
 
-static Tensor wrapped_scalar_tensor(
+at::Tensor scalar_to_tensor_default_dtype(
+    Scalar s,
+    const Device device = at::kCPU) {
+  if (s.isFloatingPoint()) {
+    return at::scalar_tensor(
+        s, at::device(device).dtype(at::get_default_dtype()));
+  } else if (s.isBoolean()) {
+    return at::scalar_tensor(s, at::device(device).dtype(at::kBool));
+  } else if (s.isComplex()) {
+    return at::scalar_tensor(
+        s, at::device(device).dtype(at::get_default_complex_dtype()));
+  } else {
+    AT_ASSERT(s.isIntegral(false));
+    return at::scalar_tensor(s, at::device(device).dtype(at::kLong));
+  }
+}
+
+// `use_default_dtype` is a bit of a hack because torch.where doesn't support type promotion, but
```

Comment:
Wow this comment really dives in!

Is this function only intended to be used by torch.where? A sentence upfront about its intended use would be helpful. In particular, if torch.where supported type promotion properly would you expect us to get rid of this function, and should this comment tell developers not to use this function?

Reply:
ya, I can add a sentence upfront summarizing.